proc/api.thrift

```diff
@@ -0,0 +1,40 @@
+service Proc {
+  
+  exception InvalidRequest {
+    1: optional string message
+  }
+
+  exception ScootTooBusy {
+    1: optional i64 retryAfterMs
+  }
+  
+  enum JobType {
+    IRON_TESTS=1,
+    DEFAULT=2,
+  }
+
+  struct Command {
+    1: list<string> cmds
+  }
+
+  struct Task {
+    1: optional string id
+    2: required Command command,
+    3: optional string snapshotId,
+    4: optional string contextId
+  }
+
+  struct Job {
+    1: required string id,
+    2: required list<Task> tasks
+  }
+
+  Job RunJob(
+    1: optional string contextId,
```

Comment:
So my thoughts were:
- Job contextId is likely to be the build you are running against, this is more for future multi Job scheduling and taking advantage of Data Locality.  I agree that we know all the tasks will benefit from the same Data Locality. 
- Task contextId is additional info like target affinity in iron tests which will help us schedule more effectively at the per task level.  I.e. I'm using 10 machines to run this Job which have github commit data locality, and running task A&B on the same machine will give me added benefits since they build tests running similar targets.